utils/aws/route53/__init__.py

```diff
@@ -0,0 +1,318 @@
+import re
+
+
+octal_re = re.compile(r'\\(\d\d\d)')
+
+
+def _octal_replace(s):
+    """
+    Replace octal characters with the actual ASCII character
+
+    See http://docs.aws.amazon.com/Route53/latest/DeveloperGuide/
+
+    :param s: source string
+    :type s: str
+    :return updated string
+    :rtype str
+    """
+    return octal_re.sub(lambda m: chr(int(m.group(1), 8)), s)
+
+
+class DuplicateException(Exception):
+    """
+    Duplicate items exception
+
+    :param item: thing that is being added
+    :type item: any
+    :param item: thing that the item is being added to
+    :type item: any
+    """
+    def __init__(self, item, to):
+        msg = f"Refusing to add duplicate {item} to {to}"
+        super(DuplicateException, self).__init__(msg)
+
+
+class InvalidRecordData(Exception):
+    """
+    Invalid DNS record data exception
+
+    :param msg: error message
+    :type msg: str
+    """
+    def __init__(self, msg):
+        super(InvalidRecordData, self).__init__(msg)
+
+
+class InvalidRecordType(Exception):
+    """
+    Unsupported record type
+
+    :param msg: error message
+    :type msg: str
+    """
+    def __init__(self, msg):
+        super(InvalidRecordType, self).__init__(msg)
+
+
+class State(object):
+    """
+    State represents a state across multiple accounts
+
+    :param name: unique name that describes the state
+    :type name: str
+    """
+    def __init__(self, name):
+        self._name = name
+        self._accounts = []
+
+    @property
+    def name(self):
+        """Returns the name of the state"""
+        return self._name
+
+    @property
+    def accounts(self):
+        """
+        Returns the accounts within the state
+
+        :return: mapping of accounts by name
+        :rtype: dict of (str, Account)
+        """
+        return {a.name: a for a in self._accounts}
```

Comment:
It is being used in `Account.get_account(name)`. I agree it's kind of unneeded but the other classes also work this way so if you have no strong objection I'd leave it as is

Reply:
I just noticed the other comments related to this. I'll look into refactoring according to your suggestion

Reply:
I understand the overhead but I think it will be minimal in this case. I did look at refactoring it anyway and found I would then need to loop over the list in `get_account` to find the name of the account with the desired name, instead of retrieving it by key. I'm not sure which is more processing, but I didn't see a great incentive to go either way. With that in mind I'll leave it as is for now and we can revisit later if this causes problems.